lib/models/report.dart

```diff
@@ -108,8 +108,14 @@ class Report {
     country = json['country'];
     nuts3 = json['nuts_3'];
     nuts2 = json['nuts_2'];
+  }
+
+  static Future<Report> fromJsonAsync(Map<dynamic, dynamic> json) async{
+    var report = Report._internal(json);
+
+    await report._initializeDisplayCity();
```

Comment:
By having it async it has time to load the displayCity. Before this, sometimes it was null randomly